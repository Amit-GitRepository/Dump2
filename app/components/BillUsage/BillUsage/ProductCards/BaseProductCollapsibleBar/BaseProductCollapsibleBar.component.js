import CircularAccordianButton from './CircularAccordianButton.component';
import Collapsible from 'react-native-collapsible';
import LinearGradient from 'react-native-linear-gradient';
import noop from 'lodash/noop';
import Proptypes from 'prop-types';
import React, {Component} from 'react';
import styles from './BaseProductCollapsibleBar.style';
import {AmountText, Checkbox, Icon, ISText, RadioButton} from '../../../../Shared';
import {colors} from '../../../../../themes/constants.styles';
import {PRODUCT_STATUS} from '../../../../../config/billUsage.config';
import {shadowActiveStyle, shadowInactiveStyle} from '../../../../../themes/application.styles';
import {translate} from '../../../../../language/i18n/helper';
import {View} from 'react-native';

class BaseProductCollapsibleBar extends Component {

  getTypeDetails= (type) => {
    const typeMap = {
      POSTPAID: {
        iconName: 'icon-true-move',
        sectionColor: colors.SECONDARY_TMOVE,
        sectionTextColor: colors.SECONDARY_TMOVE
      },
      PREPAID: {
        iconName: 'icon-true-move',
        sectionColor: colors.SECONDARY_TMOVE,
        sectionTextColor: colors.SECONDARY_TMOVE
      },
      ONLINE: {
        iconName: 'icon-true-online',
        sectionColor: colors.SECONDARY_TONLINE,
        sectionTextColor: colors.SECONDARY_TONLINE
      },
      VISION: {
        iconName: 'icon-true-visions',
        sectionColor: colors.SECONDARY_TVISION,
        sectionTextColor: colors.SECONDARY_TVISION
      },
      SMARTCHOICE: {
        iconName: 'icon-true-smart-choice',
        sectionColor: [colors.SECONDARY_TSMARTCHOICE_START, colors.SECONDARY_TSMARTCHOICE_MIDDLE, colors.SECONDARY_TSMARTCHOICE_END],
        sectionTextColor: colors.SECONDARY_TVISION
      }
    };
    return typeMap[type] || {};
  }

  getStatusTextAndColors = (statusCode, sectionColor, sectionTextColor, isNotPrepaid) => {
    let iconSectionColor = colors.PRIMARY_DISABLED_BG_TEXT;
    let textSectionColor = colors.PRIMARY_DISABLED_BG_TEXT;
    let statusText = '';

    switch (statusCode) {
    case PRODUCT_STATUS.ACTIVE: {
      iconSectionColor = sectionColor;
      textSectionColor = (isNotPrepaid) ? sectionTextColor :  colors.PRIMARY_DISABLED_BG_TEXT;
      break;
    } case PRODUCT_STATUS.SUSPEND: {
      iconSectionColor = (isNotPrepaid) ? colors.PRIMARY_DISABLED_BG_TEXT :  sectionTextColor;
      textSectionColor = (isNotPrepaid) ? colors.PRIMARY_DISABLED_BG_TEXT :  sectionTextColor;
      statusText = (isNotPrepaid) ? translate('BILLS_USAGE_SUSPENDED') : '';
      break;
    } case PRODUCT_STATUS.CANCEL: {
      statusText = translate('BILLS_USAGE_CANCELLED');
      break;
    }
    }
    return {
      iconSectionColor,
      statusText,
      textSectionColor
    };
  }

  render () {
    const {type, statusCode, leftTitleText, leftDetailText, rightTitleText, amountText, children, onCollapseToggle, isCollapsed, onChecked, isChecked, leftTitleTextStyle, checkboxDisabled} = this.props;
    const isNotPrepaid = !(type === 'PREPAID');
    const {iconName, sectionColor, sectionTextColor} = this.getTypeDetails(type);
    const {statusText, iconSectionColor, textSectionColor} = this.getStatusTextAndColors(statusCode, sectionColor, sectionTextColor, isNotPrepaid);
    const leftSectionStyles = (isNotPrepaid) ? {} : styles.flipColumn;
    const leftStyle = (statusText) ? styles.statusLeftSection : styles.leftSection;
    const rightStyle = (isNotPrepaid) ? [styles.rightSection, styles.notPrepaidRightSection] : styles.rightSection;
    const rightStatusStyle = (statusText) ? [rightStyle, styles.statusRightSection] : rightStyle;
    const leftDetailStyle = (isNotPrepaid) ? [styles.detailText, {color: textSectionColor}] : [styles.leftText, leftTitleTextStyle];
    const containerShadowStyle = isCollapsed ? shadowInactiveStyle : shadowActiveStyle;
    const productCardHeight = isNotPrepaid ? styles.postpaidProductCard : styles.prepaidProductCard;
    const selectButton = (isNotPrepaid) ? <Checkbox style={styles.checkbox} isChecked={isChecked} onChange={onChecked} disabled={checkboxDisabled} />
      : <RadioButton style={styles.checkbox} isSelected={isChecked} onRadioSelect={onChecked} />;
    return (
      <View style={[styles.container, containerShadowStyle]}>
        <View style={styles.subContainer}>
          <View style={styles.cardContainer}>
            <LinearGradient colors={Array.isArray(sectionColor) ? iconSectionColor :  [iconSectionColor, iconSectionColor]} start={{x: 0, y: 1}} end={{x: 1, y: 1}} style={styles.iconContainer}>
              <Icon name={iconName} style={styles.iconStyles}/>
            </LinearGradient>
            <View style={[styles.detailContainer, productCardHeight]}>
              <View style={styles.infoSection}>
                <View style={leftStyle}>
                  {statusText ? <ISText style={styles.statusText} type='BOLD'>{statusText}</ISText> : null}
                  <View style={leftSectionStyles}>
                    <ISText style={styles.leftTitleText} type='BOLD'>{leftTitleText}</ISText>
                    <ISText style={leftDetailStyle} type='BOLD'>{leftDetailText}</ISText>
                  </View>
                </View>
                <View style={rightStatusStyle}>
                  {(!isNotPrepaid) ? <ISText type='BOLD' style={styles.rightDetailText}>{rightTitleText}</ISText> : null}
                  <View style={styles.amountSection}>
                    <AmountText value={amountText}/>
                    {selectButton}
                  </View>
                </View>
              </View>
              <CircularAccordianButton onClick={onCollapseToggle} isCollapsed={isCollapsed} style={styles.accordianButton}/>
            </View>
          </View>
          <Collapsible collapsed={isCollapsed}>
            <View style={styles.childContainer}>
              {children}
            </View>
          </Collapsible>
        </View>
      </View>
    );
  }
}
BaseProductCollapsibleBar.defaultProps = {
  leftTitleText: '',
  leftDetailText: '',
  rightTitleText: '',
  amountText: 0,
  statusCode: 'ACTIVE',
  type: 'PREPAID',
  isCollapsed: true,
  onCollapseToggle: noop,
  onChecked: noop,
  isChecked: false,
  leftTitleTextStyle: {},
  checkboxDisabled: false
};
BaseProductCollapsibleBar.propTypes = {
  leftTitleText: Proptypes.string,
  leftDetailText: Proptypes.string,
  rightTitleText: Proptypes.string,
  amountText: Proptypes.number,
  children: Proptypes.node,
  statusCode: Proptypes.string,
  type: Proptypes.oneOf(['PREPAID', 'POSTPAID', 'ONLINE', 'VISION', 'SMARTCHOICE']),
  onCollapseToggle: Proptypes.func,
  isCollapsed: Proptypes.bool,
  onChecked: Proptypes.func,
  isChecked: Proptypes.bool,
  leftTitleTextStyle: Proptypes.object,
  checkboxDisabled: Proptypes.bool
};
export default BaseProductCollapsibleBar;
