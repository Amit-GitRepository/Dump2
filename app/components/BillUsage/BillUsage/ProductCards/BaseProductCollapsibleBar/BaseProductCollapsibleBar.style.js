import {colors, fonts} from '../../../../../themes/constants.styles';
import {shadowActiveStyle} from '../../../../../themes/application.styles';

const sectionStyles = {
  alignSelf: 'stretch',
  flexDirection: 'column',
  justifyContent: 'flex-start',
  paddingTop: 10
};

export default {
  container: {
    marginVertical: 5,
    backgroundColor: colors.TRANSPARENT,
    elevation: 2,
    borderRadius: 8
  },
  subContainer: {
    overflow: 'hidden',
    borderWidth: 1,
    borderLeftWidth: 0,
    borderRadius: 8,
    borderColor: colors.PRIMARY_DISABLED_BG_TEXT,
    backgroundColor: colors.PRIMARY_BG_TEXT_CONTRAST
  },
  cardContainer: {
    flexDirection: 'row'
  },
  iconContainer: {
    marginVertical: -1,
    alignItems: 'center',
    justifyContent: 'center',
    width: 32
  },
  iconStyles: {
    color: colors.PRIMARY_BG_TEXT_CONTRAST,
    fontSize: fonts.FONT_SIZE_XXL,
    backgroundColor: colors.TRANSPARENT
  },
  detailContainer: {
    flexGrow: 1,
    flexDirection: 'column'
  },
  infoSection: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'flex-start',
    height: '100%'
  },
  leftSection: {...sectionStyles, ...{
    flexGrow: 1,
    paddingLeft: 10
  }},
  statusLeftSection: {...sectionStyles, ...{
    flexGrow: 1,
    paddingLeft: 10,
    paddingTop: 1
  }},
  rightSection: {...sectionStyles, ...{
    paddingRight: 15
  }},
  statusRightSection: {...sectionStyles, ...{
    paddingRight: 15,
    paddingTop: 14
  }},
  amountSection: {
    flexDirection: 'row',
    alignSelf: 'flex-end',
    justifyContent: 'space-between',
    alignItems: 'flex-start',
    paddingRight: 5,
    marginBottom: -5
  },
  checkbox: {
    paddingLeft: 10,
    alignSelf: 'center',
    marginBottom: 0
  },
  leftTitleText: {
    fontSize: fonts.FONT_SIZE_MEDIUM,
    marginBottom: -5
  },
  detailText: {
    fontSize: fonts.FONT_SIZE_SMALL,
    marginBottom: -5
  },
  rightDetailText: {
    fontSize: fonts.FONT_SIZE_SMALL,
    marginBottom: -6,
    alignSelf: 'flex-end',
    paddingRight: 38
  },
  flipColumn: {
    flexDirection: 'column-reverse'
  },
  accordianButton: {
    alignSelf: 'center',
    position: 'absolute',
    bottom: 0,
    transform: [{translateX: -16}] // 32/2 :: 32 is the size of left icon
  },
  childContainer: {
    flexGrow: 1,
    backgroundColor: colors.PRIMARY_BG_SEPARATOR,
    borderColor: colors.TRANSPARENT,
    ...shadowActiveStyle
  },
  leftText: {
    fontSize: fonts.FONT_SIZE_SMALL,
    marginBottom: -6,
    paddingBottom: 5,
    alignSelf: 'flex-start'
  },
  statusText: {
    fontSize: fonts.FONT_SIZE_SMALL,
    marginBottom: -5,
    color: colors.PRIMARY_TEXT_TAB_LABEL
  },
  prepaidDetailText: {
    color: colors.PRIMARY_DISABLED_BG_TEXT,
    marginBottom: -5
  },
  notPrepaidRightSection: {
    paddingTop: 5
  },
  prepaidProductCard: {
    height: 78
  },
  postpaidProductCard: {
    height: 68
  }
};
