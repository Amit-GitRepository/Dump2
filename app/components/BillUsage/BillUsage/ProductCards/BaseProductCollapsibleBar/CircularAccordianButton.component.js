import noop from 'lodash/noop';
import Proptypes from 'prop-types';
import React, {Component} from 'react';
import semiCircle from '../../../../../assets/semiCircle.png';
import styles from './CircularAccordianButton.style';
import {Banner, Touchable} from '../../../../Shared';

class CircularAccordianButton extends Component {
  render () {
    const {isCollapsed, style, onClick} = this.props;
    const semiCircleImage = isCollapsed ? semiCircle : semiCircle;
    const accordionStyle = styles.accordionStyle;
    return (
      <Touchable borderless={true} onPress={onClick} style={[styles.container, style]}>
        <Banner source={semiCircleImage} style={accordionStyle} />
      </Touchable>
    );
  }
}
CircularAccordianButton.defaultProps = {
  isCollapsed: true,
  onClick: noop,
  style: {}
};
CircularAccordianButton.propTypes = {
  isCollapsed: Proptypes.bool,
  onClick: Proptypes.func,
  style: Proptypes.object
};

export default CircularAccordianButton;
