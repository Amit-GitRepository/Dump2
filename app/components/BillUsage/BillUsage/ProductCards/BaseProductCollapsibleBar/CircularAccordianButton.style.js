import {colors, fonts} from '../../../../../themes/constants.styles';

export default {
  container: {
    height: 21,
    width: 40,
    position: 'absolute',
    bottom: 0,
    transform: [{translateX: -16}] // 32/2 :: 32 is the size of left icon
  },
  closedContainer: {
    borderWidth: 0
  },
  accordionStyle: {
    height: 21,
    width: 40
  },
  circularContainer: {
    height: 20,
    width: 40,
    marginBottom: -1,
    borderTopLeftRadius: 40,
    borderTopRightRadius: 40,
    borderWidth: 1,
    borderColor: colors.FRENCH_GREY,
    backgroundColor: colors.GRAY_NURSE,
    alignItems: 'center',
    justifyContent: 'center'
  },
  collapsedArrowStyle: {
    transform: [{rotate: '180deg'}]
  },
  arrowStyle: {
    color: colors.PRIMARY_ACTIONABLE,
    fontSize: fonts.FONT_SIZE_SMALL
  }
};
