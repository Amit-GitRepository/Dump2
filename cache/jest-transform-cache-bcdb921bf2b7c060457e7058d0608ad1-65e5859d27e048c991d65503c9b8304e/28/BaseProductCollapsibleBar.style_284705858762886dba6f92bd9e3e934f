Object.defineProperty(exports,"__esModule",{value:true});var cov_1jn3v6ij94=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillUsage/BillUsage/ProductCards/BaseProductCollapsibleBar/BaseProductCollapsibleBar.style.js',hash='79b7006c93f546e24ec555b7c375e8b977038d25',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillUsage/BillUsage/ProductCards/BaseProductCollapsibleBar/BaseProductCollapsibleBar.style.js',statementMap:{'0':{start:{line:4,column:22},end:{line:9,column:1}}},fnMap:{},branchMap:{},s:{'0':0},f:{},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _extends=Object.assign||function(target){for(var i=1;i<arguments.length;i++){var source=arguments[i];for(var key in source){if(Object.prototype.hasOwnProperty.call(source,key)){target[key]=source[key];}}}return target;};var _constants=require('../../../../../themes/constants.styles');
var _application=require('../../../../../themes/application.styles');

var sectionStyles=(cov_1jn3v6ij94.s[0]++,{
flexDirection:'column',
justifyContent:'flex-start',
alignItems:'flex-start',
paddingTop:10});exports.default=


{
container:{
marginVertical:5,
backgroundColor:_constants.colors.TRANSPARENT},

subContainer:{
overflow:'hidden',
borderWidth:1,
borderLeftWidth:0,
borderRadius:10,
borderColor:_constants.colors.PRIMARY_DISABLED_BG_TEXT,
backgroundColor:_constants.colors.PRIMARY_BG_TEXT_CONTRAST},

cardContainer:{
flexDirection:'row'},

iconContainer:{
marginVertical:-1,
alignItems:'center',
justifyContent:'center'},

iconStyles:{
color:_constants.colors.PRIMARY_BG_TEXT_CONTRAST,
fontSize:_constants.fonts.FONT_SIZE_XL,
backgroundColor:_constants.colors.TRANSPARENT},

detailContainer:{
flexGrow:1,
flexDirection:'column'},

infoSection:{
flexDirection:'row',
justifyContent:'space-between',
alignItems:'flex-start'},

leftSection:_extends({},sectionStyles,{
paddingLeft:10}),

statusLeftSection:_extends({},sectionStyles,{
paddingLeft:10,
paddingTop:1}),

rightSection:_extends({},sectionStyles,{
paddingRight:10}),

statusRightSection:_extends({},sectionStyles,{
paddingRight:10,
paddingTop:20}),

amountSection:{
flexDirection:'row',
alignSelf:'stretch',
justifyContent:'space-between',
paddingRight:10,
paddingTop:5},

checkbox:{
paddingLeft:10,
alignSelf:'center',
marginBottom:4},

leftTitleText:{
fontSize:_constants.fonts.FONT_SIZE_MEDIUM,
marginBottom:-5},

detailText:{
fontSize:_constants.fonts.FONT_SIZE_SMALL,
marginBottom:-5},

rightDetailText:{
fontSize:_constants.fonts.FONT_SIZE_SMALL,
marginBottom:-10},

flipColumn:{
flexDirection:'column-reverse'},

accordianButton:{
alignSelf:'center'},

childContainer:_extends({
flexGrow:1,
backgroundColor:_constants.colors.PRIMARY_BG_SEPARATOR,
borderColor:_constants.colors.TRANSPARENT},_application.shadowActiveStyle),


leftText:{
paddingBottom:5},

statusText:{
fontSize:_constants.fonts.FONT_SIZE_SMALL,
marginBottom:-5,
color:_constants.colors.PRIMARY_TEXT_TAB_LABEL}};