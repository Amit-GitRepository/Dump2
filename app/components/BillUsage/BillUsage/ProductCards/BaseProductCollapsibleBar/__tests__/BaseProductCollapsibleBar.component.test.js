import BaseProductCollapsibleBar from '../BaseProductCollapsibleBar.component';
import React from 'react';
import ReactSixteenAdapter from 'enzyme/build/adapters/ReactSixteenAdapter';
import renderer from 'react-test-renderer'; // Enzyme adapter for React 16
import {configure, shallow} from 'enzyme';

configure({adapter: new ReactSixteenAdapter()});

describe('BaseProductCollapsibleBar component', () => {
  it('renders correctly', () => {
    const component = renderer.create(<BaseProductCollapsibleBar />).toJSON();
    expect(component).toMatchSnapshot();
  });
  it('should shallow render with default props', () => {
    const wrapper = shallow(<BaseProductCollapsibleBar />);
    expect(wrapper).toBeDefined();
  });
  it('should have isCollapsed state as false', () => {
    const wrapper = shallow(<BaseProductCollapsibleBar />);
    expect(wrapper.state.isCollapsed).toBeFalsy();
  });
});
