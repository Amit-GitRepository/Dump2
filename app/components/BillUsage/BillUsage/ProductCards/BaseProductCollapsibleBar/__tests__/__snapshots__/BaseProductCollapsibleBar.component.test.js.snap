// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseProductCollapsibleBar component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderRadius": 8,
        "elevation": 2,
        "marginVertical": 5,
      },
      Object {
        "shadowColor": "#303C46",
        "shadowOffset": Object {
          "height": 2,
          "width": 1,
        },
        "shadowOpacity": 0.4,
        "shadowRadius": 1,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#B9C1C9",
        "borderLeftWidth": 0,
        "borderRadius": 8,
        "borderWidth": 1,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <BVLinearGradient
        colors={
          Array [
            4294687774,
            4294687774,
          ]
        }
        end={
          Object {
            "x": 1,
            "y": 1,
          }
        }
        locations={null}
        start={
          Object {
            "x": 0,
            "y": 1,
          }
        }
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginVertical": -1,
            "width": 32,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Object {
                "backgroundColor": "transparent",
                "color": "#FFFFFF",
                "fontSize": 30,
              },
              Object {
                "fontFamily": "iservice",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </BVLinearGradient>
      <View
        style={
          Array [
            Object {
              "flexDirection": "column",
              "flexGrow": 1,
            },
            Object {
              "height": 78,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "height": "100%",
              "justifyContent": "space-between",
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "stretch",
                "flexDirection": "column",
                "flexGrow": 1,
                "justifyContent": "flex-start",
                "paddingLeft": 10,
                "paddingTop": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "column-reverse",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#303C46",
                    "fontFamily": "SukhumvitSet-Bold",
                    "fontSize": 18,
                    "marginBottom": -5,
                  }
                }
              >
                
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#303C46",
                    },
                    Array [
                      Object {
                        "alignSelf": "flex-start",
                        "fontSize": 14,
                        "marginBottom": -6,
                        "paddingBottom": 5,
                      },
                      Object {},
                    ],
                    Object {
                      "fontFamily": "SukhumvitSet-Bold",
                    },
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignSelf": "stretch",
                "flexDirection": "column",
                "justifyContent": "flex-start",
                "paddingRight": 15,
                "paddingTop": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "flex-end",
                  "backgroundColor": "transparent",
                  "color": "#303C46",
                  "fontFamily": "SukhumvitSet-Bold",
                  "fontSize": 14,
                  "marginBottom": -6,
                  "paddingRight": 38,
                }
              }
            >
              
            </Text>
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "alignSelf": "flex-end",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": -5,
                  "paddingRight": 5,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "flex-end",
                      "flexDirection": "row",
                    },
                    Object {},
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#303C46",
                      },
                      Array [
                        Object {
                          "fontSize": 16,
                          "paddingBottom": 3,
                          "paddingRight": 3,
                        },
                        Object {},
                      ],
                      Object {
                        "fontFamily": "SukhumvitSet-Bold",
                      },
                    ]
                  }
                >
                  ฿
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#303C46",
                      },
                      Array [
                        Object {
                          "fontSize": 22,
                        },
                        Object {},
                      ],
                      Object {
                        "fontFamily": "SukhumvitSet-Bold",
                      },
                    ]
                  }
                >
                  0
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#303C46",
                      },
                      Array [
                        Object {
                          "color": "#68747E",
                          "fontSize": 14,
                          "marginBottom": -1,
                          "paddingBottom": 5,
                        },
                        Object {},
                      ],
                      Object {
                        "fontFamily": "SukhumvitSet-Bold",
                      },
                    ]
                  }
                >
                  .
                  00
                </Text>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignSelf": "center",
                    "marginBottom": 0,
                    "opacity": 1,
                    "paddingLeft": 10,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "borderColor": "#F93E60",
                        "borderRadius": 11,
                        "borderWidth": 1,
                        "height": 22,
                        "justifyContent": "center",
                        "width": 22,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "#E9E9EA",
                          "borderRadius": 6,
                          "height": 12,
                          "width": 12,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "bottom": 0,
              "height": 21,
              "opacity": 1,
              "position": "absolute",
              "transform": Array [
                Object {
                  "translateX": -16,
                },
              ],
              "width": 40,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={Object {}}
            >
              <Image
                getSize={[Function]}
                isBackground={false}
                onLayout={[Function]}
                resizeMode="contain"
                source={
                  Object {
                    "testUri": "../../../app/assets/semiCircle.png",
                  }
                }
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "height": undefined,
                      "width": undefined,
                    },
                    undefined,
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      collapsable={undefined}
      pointerEvents="none"
      style={
        Object {
          "height": 0,
          "overflow": "hidden",
        }
      }
    >
      <View
        collapsable={undefined}
        onLayout={[Function]}
        style={Object {}}
      >
        <View
          style={
            Object {
              "backgroundColor": "#E9E9EA",
              "borderColor": "transparent",
              "flexGrow": 1,
              "shadowColor": "#303C46",
              "shadowOffset": Object {
                "height": 2,
                "width": 1,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 1,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
